### ISSUE TITLE
Invalid MA0047 reported

### ISSUE THREAD

**Issue by CheloXL** [meziantou/Meziantou.Analyzer#421]
**Context**
Rule identifier: _MA0047_
Target Framework: _.NET 7.0_
Language version: _11.0_

Sometimes the analyzer reports the above issue, but the class is inside a namespace. This seems to happen when a file contains multiple clases (more than 2).

The following code is reporting MA0047 in `PermitByPlateQueryHandler`:
````c#
namespace Api.Backend.Queries
{
	public sealed record PermitByPlateQuery(string Plate) : IQuery<Result<PermitByPlateQueryResult>>;

	public sealed record PermitByPlateQueryResult 
	{
		public required DateTime ValidFrom { get; init; }
		public required DateTime? ValidTo { get; init; }
	}

	public sealed class PermitByPlateQueryHandler : BaseQuery, IQueryHandler<PermitByPlateQuery, Result<PermitByPlateQueryResult>>
	{
	}
}
````
If I delete `PermitByPlateQueryResult`, the error is gone. Deleting `PermitByPlateQuery` seems to not change anything.

**Comment by meziantou** [comment]
Sorry, I do not reproduce the error on my side with the latest version.
Could you provide a zip or a link to a GitHub repository containing a solution that reproduce the error?

**Comment by CheloXL** [comment]
Unfortunatelly no. It's an internal project. I'm also on the latest version. Not sure how else can I help you to identify where the problem is.

**Comment by CheloXL** [comment]
Yes. But I've found that the error appears when using Rider. In VS I only get MA0048. In Rider I get both MA0047 and MA0048.
Not sure how Rider uses VS analyzers, but this is the first time I see different results in VS/Rider.
I'm attaching the project I'm using to test. Nothing complicated.

[ConsoleAppCore.zip](https://github.com/meziantou/Meziantou.Analyzer/files/10095288/ConsoleAppCore.zip)

**Comment by meziantou** [comment]
MA0047 is a warning, so it should be visible using `dotnet build`. If it's not the case, I can't do anything to help you.

If it only occurs in Rider (and maybe Resharper), you can open a ticket on their issue tracker.
